sherpa/conftest.py

```diff
@@ -0,0 +1,84 @@
+#
+#  Copyright (C) 2016  Smithsonian Astrophysical Observatory
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+import pytest
+from sherpa.utils import SherpaTestCase
+
+TEST_DATA_OPTION = "--test-data"
+
+
+def pytest_addoption(parser):
+    parser.addoption("-D", TEST_DATA_OPTION, action="store",
+        help="Alternative location of test data files")
+
+
+# Whilelist of known warnings. One can associate different warning messages to the same warning class
+known_warnings = {
+    DeprecationWarning:
+        ["unorderable dtypes; returning scalar but in the future this will be an error",
+         "Non-string object detected for the array ordering. Please pass in 'C', 'F', 'A', or 'K' instead",
+         "using a non-integer number instead of an integer will result in an error in the future", ],
+    UserWarning:
+        ["does not have write permission.  Changing to read-only mode.", ],
```

Comment:
As I mentioned in https://github.com/sherpa/sherpa/issues/123#issuecomment-170118512 I strongly believe we should change the code to explicitly set "mode=read-only" rather than "read-write", so we can hopefully remove this warning from the set.